formidable/forms/validations/presets.py

```diff
@@ -179,6 +203,14 @@ class ConfirmationPresets(Presets):
     description = _("Ensure both fields are identical")
     default_message = _("{left} is not equal to {right}")
 
+    def __init__(self, arguments, fields=None, message=None):
+        super(ConfirmationPresets, self).__init__(
+            arguments, fields=fields, message=message)
+        right_arg = self.arguments['right']
+        if not right_arg.field_id and fields:
```

Comment:
Instead of doing this kind of check "manually" I would like to introduce this automatically when needed.

We can imagine to declare the cast event in the Argument Object itself at the its declaration.

```
left = PresetFieldArgument(
￼              _('Reference'), help_text=_('field to compare'), order=0
)
right = PresetFieldOrValueArgument(
             _('Compare to'), help_text=_('compare with'), cast_value_with='left', order=1
 )
```

Du coup dans l'instantiation d'un presets on aurait systèmatiquement ça:

```

def __init__(self, arguments....):
      # your arguments dict transformiation here...
      for key, arg in self._declared_arguments.items:      
             if args.cast_value_with
                    arguments[key].cast_value(arguments[args.cast_value_with])
         

```

Ca permets d'avoir une méthode générique poru caster quand c'est déclarer plutot que de faire systèmatiquement ça dans toutes les prochains presets.